src/parse.rs

```diff
@@ -2126,8 +1957,9 @@ fn scan_reference<'a, 'b>(tree: &'a Tree<Item>, text: &'b str, cur: TreePointer)
         TreePointer::Valid(cur_ix) => cur_ix,
     };
     let start = tree[cur_ix].item.start;
+    let tail = &text.as_bytes()[start..];
     
-    if text[start..].starts_with("[]") {
+    if tail.len() >= 2 && tail[0] == b'[' && tail[1] == b']' {
```

Comment:
Did you try `starts_with(b"[]")` here (and in related places)? It seems clearer and I'd be surprised if it generated worse code.

Reply:
Good point! Godbolt confirms that the two generate similar code. Will change.